DM: sync drop partition failed due to "[ddl:1505]Partition management on a not partitioned table is not possible"

### What did you do?

From https://asktug.com/t/topic/1029420/2

Create one list partition then drop one partition in upstream.
```
CREATE TABLE employees11 (
id INT NOT NULL,
hired DATE NOT NULL DEFAULT ‘1970-01-01’,
store_id INT,
PRIMARY KEY (id,store_id)
)
PARTITION BY LIST (store_id) (
PARTITION pNorth VALUES IN (1, 2, 3, 4, 5),
PARTITION pEast VALUES IN (6, 7, 8, 9, 10),
PARTITION pWest VALUES IN (11, 12, 13, 14, 15),
PARTITION pCentral VALUES IN (16, 17, 18, 19, 20)
) ;

ALTER TABLE employees11 DROP PARTITION pEast;
```

### What did you expect to see?

No error report by dm.

### What did you see instead?

DM reports the following errors.
```
[2024/07/17 16:13:27.422 +08:00] [ERROR] [subtask.go:359] [“unit process error”] [subtask=testdm8] [unit=Sync] [“error information”="ErrCode:44006 ErrClass:"schema-tracker" ErrScope:"internal" ErrLevel:
"high" Message:"startLocation: [position: (mysql-bin.000001, 56898730), gtid-set: 00000000-0000-0000-0000-000000000000:0], endLocation: [position: (mysql-bin.000001, 56898854), gtid-set: 00000000-0000-0000
-0000-000000000000:0], origin SQL: [ALTER TABLE employees11 DROP PARTITION pEast]: cannot track DDL: ALTER TABLE testdm8.employees11 DROP PARTITION pEast" RawCause:"[ddl:1505]Partition management on
a not partitioned table is not possible" Workaround:"You can use handle-error to replace or skip this DDL." "]
```

### Versions of the cluster

DM version (run `dmctl -V` or `dm-worker -V` or `dm-master -V`):
v8.1 and nightly.
```console
(paste DM version here, and you must ensure versions of dmctl, DM-worker and DM-master are same)
```
Upstream MySQL/MariaDB server version:
```console
(paste upstream MySQL/MariaDB server version here)
```
Downstream TiDB cluster version (execute `SELECT tidb_version();` in a MySQL client):
```console
(paste TiDB cluster version here)
```
How did you deploy DM: tiup or manually?
```console
(leave TiUP or manually here)
```
Other interesting information (system version, hardware config, etc):
```console
>
>
```

### current status of DM cluster (execute `query-status <task-name>` in dmctl)

```console
(paste current status of DM cluster here)
```

--- a/go.mod
+++ b/go.mod
@@ -67,8 +67,8 @@ require (
 	github.com/pingcap/failpoint v0.0.0-20240528011301-b51a646c7c86
 	github.com/pingcap/kvproto v0.0.0-20240522024016-df42997c2c57
 	github.com/pingcap/log v1.1.1-0.20240314023424-862ccc32f18d
-	github.com/pingcap/tidb v1.1.0-beta.0.20240718034632-b8bfc5fb2255
-	github.com/pingcap/tidb/pkg/parser v0.0.0-20240718034632-b8bfc5fb2255
+	github.com/pingcap/tidb v1.1.0-beta.0.20240722102833-08785b6f053c
+	github.com/pingcap/tidb/pkg/parser v0.0.0-20240722102833-08785b6f053c
 	github.com/prometheus/client_golang v1.19.1
 	github.com/prometheus/client_model v0.6.1
 	github.com/r3labs/diff v1.1.0
@@ -85,7 +85,7 @@ require (
 	github.com/swaggo/gin-swagger v1.2.0
 	github.com/swaggo/swag v1.16.3
 	github.com/syndtr/goleveldb v1.0.1-0.20210305035536-64b5b1c73954
-	github.com/tikv/client-go/v2 v2.0.8-0.20240531122021-7a74511a5241
+	github.com/tikv/client-go/v2 v2.0.8-0.20240716121331-2cd3a741ea8e
 	github.com/tikv/pd v1.1.0-beta.0.20240407022249-7179657d129b
 	github.com/tikv/pd/client v0.0.0-20240708075403-19f65c59bb89
 	github.com/tinylib/msgp v1.1.6

